### PR TITLE
Add new package: smb2

### PR DESCRIPTION
**Pull request by eezstreet**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test YOUR_PACKAGE_NAME`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` should be present and it shouldn't have any additional or disabling of rules. Just content as `{ "extends": "dtslint/dt.json" }`. If for reason the some rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]`  and not for whole package so that the need for disabling can be reviewed.
- [x] `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/smb2/tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,24 @@
+{
+    "compilerOptions": {
+		"baseUrl": "../",
```
Comment (danger-public) [inline types/smb2/tsconfig.json:3]: - :no_entry_sign: The root .editorconfig style specifies spaces for whitespace. Please use spaces in new or changed types.

File: types/smb2/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,54 @@
+// Type definitions for smb2 0.2
+// Project: https://github.com/bchelli/node-smb2
+// Definitions by: Nick Whitlock <https://github.com/eezstreet>
+// Definitions: https://github.com/eezstreet/DefinitelyTyped
+
+export = SMB2;
+
+interface SMB2Options {
+	share: string;
```
Comment (danger-public) [inline types/smb2/index.d.ts:9]: - :no_entry_sign: The root .editorconfig style specifies spaces for whitespace. Please use spaces in new or changed types.

File: types/smb2/tslint.json [hunk 3]
```diff
@@ -0,0 +1,6 @@
+{ 
+	"extends": "dtslint/dt.json",
```
Comment (danger-public) [inline types/smb2/tslint.json:2]: - :no_entry_sign: The root .editorconfig style specifies spaces for whitespace. Please use spaces in new or changed types.

**Comment by danger-public** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  **types/smb2/smb2-tests.ts#L2** - The root .editorconfig style specifies spaces for whitespace. Please use spaces in new or changed types.
  </td>
    </tr>
  </tbody>
</table>




<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against f9446b44cb566655df372e9195f13c6fdbfd6203
</p>

File: types/smb2/smb2-tests.ts [hunk 4]
```diff
@@ -0,0 +1,4 @@
+function uhOh() {
+	// I don't have any tests!
```
Comment (danger-public) [inline types/smb2/smb2-tests.ts:2]: - :no_entry_sign: The root .editorconfig style specifies spaces for whitespace. Please use spaces in new or changed types.

File: types/smb2/index.d.ts [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+
+    autoCloseTimeout?: number;
+}
+
+interface SMB2ReadFileOptions {
+    encoding: string | null;
+}
+
+declare class SMB2 {
+    constructor(options: SMB2Options);
```
Comment (peterblazejewicz) [inline types/smb2/index.d.ts:29]:
```suggestion
    constructor(options?: SMB2Options);
```
https://github.com/bchelli/node-smb2/blob/master/lib/smb2.js#L24

File: types/smb2/tslint.json [hunk 6]
```diff
@@ -0,0 +1,6 @@
...
+	"rules": {
+		"npm-naming": false
```
Comment (peterblazejewicz) [inline types/smb2/tslint.json:4]: why do you need this?

File: types/smb2/smb2-tests.ts [hunk 7]
```diff
@@ -0,0 +1,4 @@
+function uhOh() {
+    // I don't have any tests!
+    // Quick, someone give me a samba server to connect to!
+}
```
Comment (peterblazejewicz) [inline types/smb2/smb2-tests.ts:4]:
```suggestion
import SMB2 = require('smb2');

constsmb2Client = new SMB2({
  share:'\\\\000.000.000.000\\c$',
  domain:'DOMAIN',
  username:'username',
  password:'password!'
});
```
you can start with README, that `.ts` are not executed, they are compiled but do not emit.
you are not to connect to any server, please amend tests

**Review by peterblazejewicz — APPROVED** [review]

**Pull request merged**